src/Microsoft.SqlTools.ServiceLayer/QueryExecution/DataStorage/SaveAsWriterBase.cs

```diff
@@ -76,6 +76,15 @@ public int WriteRow(StorageDataReader dataReader)
         public abstract void WriteRow(IList<DbCellValue> row, IList<DbColumnWrapper> columns);
 
         /// <summary>
+        /// Not implemented, do not use.
+        /// </summary>
+        [Obsolete]
+        public void Seek(long offset)
+        {
+            throw new InvalidOperationException("SaveAs writers are meant to be written once contiguously.");
```

Comment:
I would go the NotImplemented route, but that is considered by ReSharper to be in the same boat as a "todo" comment.

I like your comment about the interfaces, but I'm not exactly sure how I'd do it. The ResultSet class would need to get a writer that implements both the Seek method and the write row methods.